source/elements/oneMKL/source/domains/spblas/data_types/create_dense_matrix.rst

```diff
@@ -0,0 +1,192 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_create_dense_matrix:
+
+create_dense_matrix
+===================
+
+Initializes a ``dense_matrix_handle_t`` object with the provided data.
+
+.. rubric:: Description and Assumptions
+
+The ``oneapi::mkl::sparse::create_dense_matrix`` function initializes the
+``dense_matrix_handle_t`` object with the provided data.
+
+In the case of buffers, the reference count of the provided buffer is
+incremented which extends the lifetime of the underlying buffer until the
+``dmhandle`` is destroyed with ``destroy_dense_matrix`` or the data is reset
+with ``set_dense_matrix_data``.
+
+In the case of USM, the object does not take ownership of the data.
+
+See :ref:`onemkl_sparse_dense_matrix_handle`.
+
+.. _onemkl_sparse_create_dense_matrix_buffer:
+
+create_dense_matrix (Buffer version)
+------------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType>
+       void create_dense_matrix (sycl::queue                                &queue,
+                                 oneapi::mkl::sparse::dense_matrix_handle_t *p_dmhandle,
+                                 std::int64_t                               num_rows,
+                                 std::int64_t                               num_cols,
+                                 std::int64_t                               ld,
+                                 layout                                     dense_layout,
+                                 sycl::buffer<dataType, 1>                  &val);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Template parameters
+
+   dataType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   p_dmhandle
+      The address of the ``p_dmhandle`` object to be initialized. Must only be
+      called on an uninitialized ``dense_matrix_handle_t`` object.
+
+   num_rows
+      Number of rows of the provided data ``val``. Must be at least 0.
+
+   num_cols
+      Number of columns of the provided data ``val``. Must be at least 0.
```

Comment:
I don't see us supporting a partially defined dense matrix handle the way we must support one for a sparse matrix handle. If the user is calling the `create_dense_matrix` API, everything being passed to it must be completely defined. I think it is safe to require `num_rows > 0`, `num_cols > 0`, and that `val` can neither be a `sycl::buffer` of length 0, nor can it be a `nullptr` in case of USM.

Reply:
Addressed in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/ad83d1c3e00c16af41cf7adbacf02ab44fd8e3e0